lib/palindrome_check.rb

```diff
@@ -1,5 +1,35 @@
 # A method to check if the input string is a palindrome.
 # Return true if the string is a palindrome. Return false otherwise.
 def palindrome_check(my_phrase)
-  raise NotImplementedError
+
+  if my_phrase == nil
+    return false
+  end
+
+  unless my_phrase.empty? || my_phrase.length == 1
+
+    index = 0
+    last = my_phrase.length - 1
+
+    while index <= last
+
+      while my_phrase[index] == ' '
```

Comment:
As further due diligence and to make your algorithm robust, ensure that index is less than or equal to last before comparing the character at index with a white space in this loop.

Since you're coding in Ruby, the character at index length will be `nil` and the loop will exit. This is not always true with other programming languages and hence, there is value in making your algorithm more robust.